src/components/transactions/Switch/SwitchAssetInput.tsx

```diff
@@ -122,11 +131,34 @@ export const SwitchAssetInput = ({
         asset.address.toLowerCase() === searchQuery
     );
 
-    setFilteredAssets(matchingAssets);
+    const isEthAddress = /^(0x)?[0-9a-fA-F]{40}$/.test(value);
```

Comment:
is there a utility function somewhere for this?

Reply:
i was not able to find, there is some in the utils package but just for decorators

Reply:
You could use `isAddress` from `ethers/lib/utils`